include/dyad/c_api/dyad_c.h

```diff
@@ -0,0 +1,34 @@
+#ifndef DYAD_C_API_DYAD_C_H
+#define DYAD_C_API_DYAD_C_H
+
+#if defined(DYAD_HAS_CONFIG)
+#include <dyad/dyad_config.hpp>
+#else
+#error "no config"
+#endif
+
+#include <dyad/core/dyad_ctx.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct dyad_ctx;
+typedef struct dyad_ctx dyad_ctx_t;
+
+__attribute__ ((__visibility__ ("default"))) int dyad_c_init (dyad_ctx_t** ctx);
+
+__attribute__ ((__visibility__ ("default"))) int dyad_c_open (dyad_ctx_t* ctx,
```

Comment:
naming it as c_open makes me think, do we need to support other semantics like `creat`, `openat` as semantics.

Also if the flag says O_DIRECTORY then it is not suppose to consume right? I feel we need to make it more specific. 

Not use open or close. But directly dyad_produce and consume for C_APIs for DYAD

Reply:
At that point, we're essentially just exposing core to the users. In fact, that's what we could do. This can be a high level C API. Essentially, it would just be a version of the wrapper that you can link into your program instead of preloading. Then, core could serve as our low level C API, providing the produce/consume interface you suggested @hariharan-devarajan.

@hariharan-devarajan @JaeseungYeom what do you think?

Reply:
This would also let me avoid using the awkward `dyad_c_...` names for these functions. The reason I chose this naming was specifically to avoid symbol conflicts between this API and core.

Reply:
> At that point, we're essentially just exposing core to the users. In fact, that's what we could do. This can be a high level C API. Essentially, it would just be a version of the wrapper that you can link into your program instead of preloading. Then, core could serve as our low level C API, providing the produce/consume interface you suggested @hariharan-devarajan.
> 
> @hariharan-devarajan @JaeseungYeom what do you think?

I agree that we can simply expose what is already there. I think eventually we want to rename wrapper to intercepter.